src/operator/elementwise_unary_op-inl.h

```diff
@@ -0,0 +1,144 @@
+  });
+}
+
+// backward function that takes input value of the op
+template<typename xpu, typename OP>
+void UnaryBackwardUseIn_(const OutputGrad& out_grad,
+                         const Input0& in_data0,
+                         const EnvArguments& env,
+                         TBlob *in_grad,
+                         OpReqType req,
+                         RunContext ctx) {
+  using namespace mxnet::op;
+  using namespace mshadow::expr;
+  mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
+  CHECK_EQ(in_grad->type_flag_, out_grad.data.type_flag_)
+    << "Unary function only support input/output with the same type";
+  CHECK_EQ(in_grad->type_flag_, in_data0.data.type_flag_)
+    << "Unary function only support input/output with the same type";
+  MSHADOW_TYPE_SWITCH(in_grad->type_flag_, DType, {
+    mshadow::Tensor<xpu, 2, DType> igrad = in_grad->FlatTo2D<xpu, DType>(s);
+    ASSIGN_DISPATCH(igrad, req,
+                    (F<OP>(in_data0.data.FlatTo2D<xpu, DType>(s)) *
+                     out_grad.data.FlatTo2D<xpu, DType>()));
+  });
+}
+
+// backward function that takes output value of the op
+template<typename xpu, typename OP>
+void UnaryBackwardUseOut_(const OutputGrad& out_grad,
+                          const OutputValue& out_value,
+                          const EnvArguments& env,
+                          TBlob *in_grad,
+                          OpReqType req,
+                          RunContext ctx) {
+  using namespace mxnet::op;
+  using namespace mshadow::expr;
+  mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
+  CHECK_EQ(in_grad->type_flag_, out_grad.data.type_flag_)
+    << "Unary function only support input/output with the same type";
+  CHECK_EQ(in_grad->type_flag_, out_value.data.type_flag_)
+    << "Unary function only support input/output with the same type";
+  MSHADOW_TYPE_SWITCH(in_grad->type_flag_, DType, {
+    mshadow::Tensor<xpu, 2, DType> igrad = in_grad->FlatTo2D<xpu, DType>(s);
+    ASSIGN_DISPATCH(igrad, req,
+                    (F<OP>(out_value.data.FlatTo2D<xpu, DType>(s)) *
+                     out_grad.data.FlatTo2D<xpu, DType>()));
+    });
+}
+
+MXNET_REGISTER_SIMPLE_OP(abs, XPU)
+.set_function(XPU::kDevMask, UnaryForward_<XPU, mshadow_op::abs>, kInplaceInOut)
+.set_gradient(XPU::kDevMask, UnaryBackwardUseIn_<XPU, mshadow_op::sign>, kInplaceOutIn)
+.describe("Take absolute value of the src");
+// sign
+MXNET_REGISTER_SIMPLE_OP(sign, XPU)
+.set_function(XPU::kDevMask, UnaryForward_<XPU, mshadow_op::sign>, kInplaceInOut)
+.set_gradient(XPU::kDevMask, UnaryBackwardUseIn_<XPU, mshadow_op::sign_grad>, kInplaceOutIn)
+.describe("Take sign value of the src");
+// round
+MXNET_REGISTER_SIMPLE_OP(round, XPU)
+.set_function(XPU::kDevMask, UnaryForward_<XPU, mshadow_op::round>, kInplaceInOut)
+.describe("Take round value of the src");
+// ceil
+MXNET_REGISTER_SIMPLE_OP(ceil, XPU)
+.set_function(XPU::kDevMask, UnaryForward_<XPU, mshadow_op::ceil>, kInplaceInOut)
```

Comment:
If gradient is not registered, they are nop and will shoot an error when backward is called.